python/cudf/cudf/core/dataframe.py

```diff
@@ -6327,6 +6328,37 @@ def nunique(self, axis=0, dropna=True):
 
         return cudf.Series(super().nunique(method="sort", dropna=dropna))
 
+    def _sample_axis_1(
+        self,
+        n: int,
+        weights: Optional[ColumnLike],
+        replace: bool,
+        random_state: np.random.RandomState,
+        ignore_index: bool,
+    ):
+        if replace:
+            raise NotImplementedError(
+                "Sample is not supported for axis 1/`columns` when"
+                "`replace=True`."
+            )
+
+        columns = self._data.names
+        if not replace and n > len(columns):
+            raise ValueError(
+                "Cannot take a larger sample than population when"
+                "`replace=False`."
+            )
+
+        sampled_column_labels = random_state.choice(
+            columns, size=n, replace=False, p=weights
```

Comment:
These are a bit inconsistent in whether or not they assume replace is False. The very first check in this function guarantees that it is False, but it probably makes sense to keep using the parameter rather than hardcoding False. on 6353. If you do want to hardcode it, though, then on 6346 you can remove the `not replace` check since it's redundant.

Reply:
In fact, this check is not needed anymore. I believe it's added because before this refactor `random.choice` wasn't used and this was added to mimic the exception thrown.

Reply:
We can also support `replace` parameter if we can make sure there's a way to deep copy the column when you gather them by label (out of scope maybe?).

Reply:
Actually NM, we can't support it because no two column can have the same name in cudf frame.